tests/unit/backend/layers/business/test_business.py

```diff
@@ -1475,5 +1475,32 @@ def test_dataset_published_at_with_replaced_dataset_ok(self):
         self.assertEqual(dataset_1.canonical_dataset.published_at, published_collection.published_at)
 
 
+class TestCollectionUtilities(BaseBusinessLogicTestCase):
```

Comment:
Add a test where multiple collection versions refers to the same dataset version.

Reply:
that is encompassed in the existing test since one dataset is updated and the other is not.

Reply:
I would mention that in the docstring, that test case is obfuscated in the `initialize_published_collection`